cg/meta/demultiplex/wipe_demultiplex_api.py

```diff
@@ -0,0 +1,160 @@
+
+        for sample in self.samples_on_flow_cell:
+            self.housekeeper_api.delete_fastq_and_spring(
+                bundle=sample.internal_id, demultiplexing_path=self.demultiplexing_run_path
+            )
+
+    def wipe_flow_cell_housekeeper(self) -> None:
+        """Wipe any presence of a flow cell in housekeeper. Including Sample sheets AND fastq-files"""
+
+        log.info(f"Housekeeper: Wiping sample sheet files with tag {self.flow_cell_name}")
+        self._wipe_sample_sheet_housekeeper()
+        log.info(f"Housekeeper: Wiping fastq and spring files related to flowcell")
+        self._wipe_fastq_and_spring_housekeeper()
+
+    def wipe_flow_cell_statusdb(self) -> None:
+        """Wipe any presence of a flow cell in status-db"""
+        if self.dry_run:
+            log.info(f"StatusDB: Would remove {self.flow_cell_name}")
+        else:
+            log.info(f"StatusDB: Wiping flowcell {self.flow_cell_name}")
+            self.status_db.delete_flowcell(flowcell_name=self.flow_cell_name)
+
+    def wipe_flow_cell_cgstats(self) -> None:
+        """Wipe ay presence of a flow cell in cgstats"""
+        from cg.apps.cgstats.crud.delete import delete_flowcell
+
+        if self.dry_run:
+            log.info(f"CGStats: Would remove {self.flow_cell_name}")
+        else:
+            delete_flowcell(manager=self.stats_api, flowcell_name=self.flow_cell_name)
+
+    def wipe_flow_cell_demultiplex_dir(self) -> None:
+        """Wipe demultiplexing directory on the server"""
+        out_dir: Path = self.demultiplexing_dir.joinpath(self.run_name.name)
+        if out_dir.exists():
+            if self.dry_run:
+                log.info(f"Hasta: Would remove the following directory {out_dir.as_posix()}")
+            else:
+                log.info(f"Hasta: Removing flow cell directory {out_dir.as_posix()}")
+                rmtree(path=out_dir)
+        else:
+            log.info(f"Hasta: No target demultiplexing directory {out_dir.as_posix()}")
+
+    def wipe_flow_cell(
+        self, cg_stats: bool, demultiplexing_dir: bool, housekeeper: bool, status_db: bool
+    ) -> None:
+        """Master command to completely wipe the presence of a flowcell in all services"""
+        if self.status_db_presence:
+            self._set_samples_on_flow_cell()
+            active_samples_on_flow_cell: List[str] = self.active_samples_on_flow_cell()
+            if active_samples_on_flow_cell:
+                log.warning(
+                    f"There are active analyses using data from this flowcell.\n"
+                    f"Before restarting the demultiplexing process - please cancel the ongoing analyses of the "
+                    f"following sample(s): \n{active_samples_on_flow_cell}"
+                )
+                raise WipeDemuxError
+        else:
+            if cg_stats:
+                self.wipe_flow_cell_cgstats()
+            elif demultiplexing_dir:
```

Comment:
I think rmtree involved in this can fail with permission errors. So thats why it should be done first, before cgstats, so that the record of the flowcell isnt removed before the files are

Reply:
agreed